hammurabi/rules/toml.py

```diff
@@ -0,0 +1,276 @@
+        >>>         TomlKeyExists(
+        >>>             name="Ensure service descriptor has stack",
+        >>>             path=Path("./service.toml"),
+        >>>             key="stack",
+        >>>             value="my-awesome-stack",
+        >>>         ),
+        >>>     )
+        >>> )
+        >>>
+        >>> pillar = Pillar()
+        >>> pillar.register(example_law)
+
+    .. warning::
+
+        Setting a value to None will result in a deleted key as per the documentation of how
+        null/nil values should be handled. More info: https://github.com/toml-lang/toml/issues/30
+
+    .. warning::
+
+        Compared to :mod:`hammurabi.rules.text.LineExists`, this rule is NOT able to add a
+        key before or after a match.
+    """
+
+
+class TomlKeyNotExists(DictKeyNotExists, SingleDocumentTomlFileRule):
+    """
+    Ensure that the given key not exists. If needed, the rule will remove a key with the
+    given name, including its value.
+
+    Example usage:
+
+        >>> from pathlib import Path
+        >>> from hammurabi import Law, Pillar, TomlKeyNotExists
+        >>>
+        >>> example_law = Law(
+        >>>     name="Name of the law",
+        >>>     description="Well detailed description what this law does.",
+        >>>     rules=(
+        >>>         TomlKeyNotExists(
+        >>>             name="Ensure outdated_key is removed",
+        >>>             path=Path("./service.toml"),
+        >>>             key="outdated_key",
+        >>>         ),
+        >>>     )
+        >>> )
+        >>>
+        >>> pillar = Pillar()
+        >>> pillar.register(example_law)
+    """
+
+
+class TomlKeyRenamed(DictKeyRenamed, SingleDocumentTomlFileRule):
+    """
+    Ensure that the given key is renamed. In case the key can not be found,
+    a ``LookupError`` exception will be raised to stop the execution. The
+    execution must be stopped at this point, because if other rules depending
+    on the rename they will fail otherwise.
+
+    Example usage:
+
+        >>> from pathlib import Path
+        >>> from hammurabi import Law, Pillar, TomlKeyRenamed
+        >>>
+        >>> example_law = Law(
+        >>>     name="Name of the law",
+        >>>     description="Well detailed description what this law does.",
+        >>>     rules=(
+        >>>         TomlKeyRenamed(
+        >>>             name="Ensure service descriptor has dependencies",
+        >>>             path=Path("./service.toml"),
+        >>>             key="development.depends_on",
+        >>>             value="dependencies",
+        >>>         ),
+        >>>     )
+        >>> )
+        >>>
+        >>> pillar = Pillar()
+        >>> pillar.register(example_law)
+    """
+
+
+class TomlValueExists(DictValueExists, SingleDocumentTomlFileRule):
+    """
+    Ensure that the given key has the expected value(s). In case the key cannot
+    be found, a ``LookupError`` exception will be raised to stop the execution.
```

Comment:
Would it make sense to create the key if it does not exist? But then how does it differ from key exist check that also sets the value. I am not sure we need both. 🤔

Reply:
`TomlValueIsSet` is set could be used here (and for yaml/json as well)

Reply:
Should be part of https://github.com/gabor-boros/hammurabi/issues/174